src/emissions-workspace/emissions.service.ts

```diff
@@ -164,146 +165,154 @@ export class EmissionsWorkspaceService {
+    try {
+      const trx = queryRunner.manager;
 
-    if (isUndefinedOrNull(plant)) {
-      throw new NotFoundException('Plant not found.');
-    }
+      const stackPipeIds = objectValuesByKey<string>('stackPipeId', params, true);
+      const unitIds = objectValuesByKey<string>('unitId', params, true);
 
-    const monitorPlans = plant.monitorPlans;
+      const plant = await this.plantRepository.getImportPlant({
+        orisCode: params.orisCode,
+        stackIds: stackPipeIds,
+        unitIds: unitIds,
+      });
 
-    if (monitorPlans.length === 0) {
-      throw new NotFoundException('Monitor plan not found.');
-    }
+      if (isUndefinedOrNull(plant)) {
+        throw new NotFoundException('Plant not found.');
+      }
 
-    if (monitorPlans.length > 1) {
-      throw new NotFoundException('Multiple active monitor plans found.');
-    }
+      const monitorPlans = plant.monitorPlans;
 
-    const reportingPeriod = await this.entityManager.findOne(ReportingPeriod, {
-      where: {
-        year: params.year,
-        quarter: params.quarter,
-      },
-    });
+      if (monitorPlans.length === 0) {
+        throw new NotFoundException('Monitor plan not found.');
+      }
 
-    if (!reportingPeriod) {
-      throw new NotFoundException('Reporting period not found.');
-    }
+      if (monitorPlans.length > 1) {
+        throw new NotFoundException('Multiple active monitor plans found.');
+      }
 
-    const monitorPlanId = monitorPlans[0].id;
-    const monitoringLocations = monitorPlans[0].locations;
+      const reportingPeriod = await trx.findOne(ReportingPeriod, {
+        where: {
+          year: params.year,
+          quarter: params.quarter,
+        },
+      });
 
-    const reportingPeriodId = reportingPeriod.id;
+      if (!reportingPeriod) {
+        throw new NotFoundException('Reporting period not found.');
+      }
 
-    const identifiers = await this.getUnifiedIdentifiers(
-      params,
-      monitoringLocations,
-      userId,
-    );
+      const monitorPlanId = monitorPlans[0].id;
+      const monitoringLocations = monitorPlans[0].locations;
 
-    // Import-28 Valid formulaIdentifiers for location
-    await this.checksService.invalidFormulasCheck(params, monitoringLocations);
+      const reportingPeriodId = reportingPeriod.id;
 
-    for (const monitorPlan of monitorPlans) {
-      await this.entityManager.query(
-        'CALL camdecmpswks.delete_monitor_plan_emissions_data_from_workspace($1, $2)',
-        [monitorPlan.id, reportingPeriodId],
+      const identifiers = await this.getUnifiedIdentifiers(
+        params,
+        monitoringLocations,
+        userId,
       );
-    }
 
-    const currentTime = currentDateTime().toISOString();
+      // Import-28 Valid formulaIdentifiers for location
+      await this.checksService.invalidFormulasCheck(params, monitoringLocations);
 
-    const importPromises = [
-      this.importDailyEmissions(
+      for (const monitorPlan of monitorPlans) {
```

Comment:
Since all previous queries target unmodified data, it may be more clear to start the transaction here:
```suggestion
    const queryRunner = this.entityManager.connection.createQueryRunner();
    await queryRunner.startTransaction();
    
    try {
      const trx = queryRunner.manager;
      
      for (const monitorPlan of monitorPlans) {
```

Reply:
Thanks for the suggestion. I definitely see the performance benefits of moving the transaction boundary to only wrap data modifications.
However, my main concern is the small window where reference data (monitor plans, reporting periods, identifiers) could potentially change between validation and transaction start, potentially causing subtle consistency issues.
What do you think?
Based on your suggestion, here is how I would implement the change: 
```
// Pre-transaction validation phase - all read-only operations
    const stackPipeIds = objectValuesByKey<string>('stackPipeId', params, true);
    const unitIds = objectValuesByKey<string>('unitId', params, true);

    const plant = await this.plantRepository.getImportPlant({
      orisCode: params.orisCode,
      stackIds: stackPipeIds,
      unitIds: unitIds,
    });

    if (isUndefinedOrNull(plant)) {
      throw new NotFoundException('Plant not found.');
    }

    const monitorPlans = plant.monitorPlans;

    if (monitorPlans.length === 0) {
      throw new NotFoundException('Monitor plan not found.');
    }

    if (monitorPlans.length > 1) {
      throw new NotFoundException('Multiple active monitor plans found.');
    }

    const reportingPeriod = await this.entityManager.findOne(ReportingPeriod, {
      where: {
        year: params.year,
        quarter: params.quarter,
      },
    });

    if (!reportingPeriod) {
      throw new NotFoundException('Reporting period not found.');
    }

    const monitorPlanId = monitorPlans[0].id;
    const monitoringLocations = monitorPlans[0].locations;

    const reportingPeriodId = reportingPeriod.id;

    const identifiers = await this.getUnifiedIdentifiers(
      params,
      monitoringLocations,
      userId,
    );

    // Import-28 Valid formulaIdentifiers for location
    await this.checksService.invalidFormulasCheck(params, monitoringLocations);

    // Transaction phase - only data modification operations
    // Since all previous queries target unmodified data, it may be more clear to start the transaction here
    const queryRunner = this.entityManager.connection.createQueryRunner();
    await queryRunner.startTransaction();
```

Reply:
I think that's a valid concern, but we account for that by only letting one person check out and modify a configuration at a single time, so it should not be a problem.

I was thinking less about the performance and more about clarity and maintenance: currently, the transaction is started, then several queries are performed using non-transaction connections. The specific queries don't pose issues in this case, but moving the transaction boundary down would serve to say "hey, modifications are happening  and it's really only safe to use the transactional connection (`trx`) from here on out".

Reply:
Sounds good. Will adjust accordingly. Thanks.